docs/examples/napari/napari_img_math.md

```diff
@@ -0,0 +1,210 @@
+function has a `Layer` return type annotation, and will add the result to the
+viewer.  So we gain a *lot* by annotating the above function with the
+appropriate `napari` types.
+
+```python
+from napari.layers import Image
+
+def image_arithmetic(
+    layerA: Image, operation: Operation, layerB: Image
+) -> Image:
+    return operation.value(layerA.data, layerB.data)
+```
+
+### the magic part
+
+ Finally, we decorate the function with `@magicgui` and tell it we'd like to have
+a `call_button` that we can click to execute the function.
+
+```python hl_lines="1"
+@magicgui(call_button="execute")
+def image_arithmetic(layerA: Image, operation: Operation, layerB: Image):
+    return operation.value(layerA.data, layerB.data)
+```
+
+That's it!  The `image_arithmetic` function is now a
+{class}`~magicgui.function_gui.FunctionGui` that can be shown, or incorporated
+into other GUIs (such as the napari GUI shown in this example)
+
+```{note}
+While [type hints](https://docs.python.org/3/library/typing.html) aren't
+always required in `magicgui`, they are recommended (see {ref}`type-inference` )...
+and they *are* required for certain things, like the `Operation(Enum)` [used here
+for the dropdown](#create-dropdowns-with-enums) and the
+{class}`napari.layers.Image <napari.layers.image.image.Image>`
+annotations that `napari` has registered with `magicgui`.
+```
+
+### create dropdowns with Enums
+
+We'd like the user to be able to select the operation (`add`, `subtract`,
+`multiply`, `divide`) using a dropdown menu.  {class}`~enum.Enum`s offer a
+convenient way to restrict values to a strict set of options, while providing
+`name: value` pairs for each of the options. Here, the value for each choice is
+the actual function we would like to have called when that option is selected.
+
+```python
+class Operation(enum.Enum):
+    add = numpy.add
+    subtract = numpy.subtract
+    multiply = numpy.multiply
+    divide = numpy.divide
+```
+
+### add it to napari
+
+When we decorated the `image_arithmetic` function above, it became a
+{class}`~magicgui.function_gui.FunctionGui`.  Napari recognizes this type,
+so we can simply add it to the napari viewer as follows:
+
+```python
+viewer.window.add_dock_widget(image_arithmetic)
+```
+
+```{caution}
+This api has changed slightly with version 0.2.0 of magicgui.  See the
+[migration guide](../../api/v0_2_0) if you are migrating from a previous version.
+```
+
+### connect event listeners for interactivity
+
+What fun is a GUI without some interactivity?  Let's make stuff happen.
+
+We connect the `image_arithmetic.reset_choices` function to the
+`viewer.layers.events.inserted/removed` event from `napari`, to make sure that the
+dropdown menus stay in sync if a layer gets added or removed from the napari
+window:
+
+```python
+viewer.layers.events.inserted.connect(image_arithmetic.reset_choices)
+viewer.layers.events.removed.connect(image_arithmetic.reset_choices)
+```
+
+````{tip}
```

Comment:
is there a difference between:

````
```{tip}
...
```
````

and

````
```{admonition}
:class: tip
...
```
````
? From the rendering on the site they seem to do the same thing?

Reply:
the main difference that I (think I) found was that you can add custom text with `{admonition}`
````
```{admonition} Here's a hot Tip!
:class: tip
```
````

but I think `{tip}` always says "Tip"